src/bessel.jl

```diff
@@ -367,6 +367,12 @@ function bessely(nu::Cint, x::Float32)
     end
     ccall((:ynf, libm), Float32, (Cint, Float32), nu, x)
 end
+function bessely(nu::Cint, x::Float16)
+    if x < 0
+        throw(DomainError(x, "`x` must be nonnegative."))
+    end
```

Comment:
This check is unnecessary, as it will be handled by the `Float32` version.

Reply:
Yup. Okay

Reply:
By the way , since all other functions throw `MethodError` for Float16 . Should i do the same here too? Or return typecasted value of Float32?

Reply:
No, I think we should define `Float16` methods for all these.

Reply:
Okay..Then . I'm closing this PR and will make a new one , once i'm done with changes. Is it fine to close this?